npm/podController.go

```diff
@@ -508,14 +535,36 @@ func (c *podController) syncAddAndUpdatePod(newPodObj *corev1.Pod) error {
 		if err = manageNamedPortIpsets(ipsMgr, cachedNpmPodObj.ContainerPorts, podKey, cachedNpmPodObj.PodIP, DeleteNamedPortIpsets); err != nil {
 			return fmt.Errorf("[syncAddAndUpdatePod] Error: failed to delete pod from named port ipset with err: %v", err)
 		}
+		addNamedPortIpsets = true
+	}
+
+	// Updating pod cache with new npmPod information
+	c.npMgr.PodMap[podKey] = newNpmPod(newPodObj)
+
+	// Add the pod to its label's ipset.
+	for _, addIPSetName := range addToIPSets {
+		log.Logf("Adding pod %s to ipset %s", newPodObj.Status.PodIP, addIPSetName)
+		if err = ipsMgr.AddToSet(addIPSetName, newPodObj.Status.PodIP, util.IpsetNetHashFlag, podKey); err != nil {
+			return fmt.Errorf("[syncAddAndUpdatePod] Error: failed to add pod to label ipset with err: %v", err)
+		}
+		// {IMPORTANT} Same as above order is assumed to be key and then key+val. NPM should only append to existing labels
+		// only after both ipsets for a given label's key value pair are added successfully
+		// (TODO) will need to remove this ordering dependency
+		addedLabel := util.GetLabelKVFromSet(addIPSetName)
+		if len(addedLabel) > 1 {
+			c.npMgr.PodMap[podKey].appendLabels(map[string]string{addedLabel[0]: addedLabel[1]}, AppendToExitingLabels)
+		}
+	}
+	// This will ensure after all labels are worked on to overwrite. This way will reduce any bugs introduced above
```

Comment:
What are possible bugs? It would be nice to update bugs case as comment.

Reply:
I added a comment. Due to the ordering issues if we not delete and add a label, this appedlabels will help override it.